docs/architecture.md

```diff
@@ -221,10 +222,20 @@ The query frontend supports caching query results and reuses them on subsequent
 
 The **ruler** is an **optional service** executing PromQL queries for recording rules and alerts. The ruler requires a database storing the recording rules and alerts for each tenant.
 
-Ruler can be scaled horizontally.
+Ruler are **stateless** and can be scaled horizontally.
 
 ### Alertmanager
 
 The **alertmanager** is an **optional service** responsible for accepting alert notifications from the [ruler](#ruler), deduplicating and grouping them, and routing them to the correct notification channel, such as email, PagerDuty or OpsGenie.
 
 The Cortex alertmanager is built on top of the [Prometheus Alertmanager](https://prometheus.io/docs/alerting/alertmanager/), adding multi-tenancy support. Like the [ruler](#ruler), the alertmanager requires a database storing the per-tenant configuration.
+
+Alertmanager are **stateful**.
+
+### Configs API
+
+The **configs API** is an **optional service** managing the configuration of Rulers and Alertmanagers.
+It provides APIs to get/set/update the rules and alertmanager configurations and store them into backend.
+Current supported backend are PostgreSQL, static files and in-memory.
+
+Configs API are **stateless** and can be scaled up and down as needed.
```

Comment:
```suggestion
Configs API is **stateless**.
```

Not sure it's worth to mention the "scaled up and down as needed" cause you don't really need to scale it up/ down (we don't expect an huge traffic on these endpoints). What you usually want is just having a couple of replicas for HA purposes.